### ISSUE TITLE
[Bug]: Opening new tab leads to error

### ISSUE THREAD

**Issue by horseinthesky** [akinsho/bufferline.nvim#441]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Opening a new tab with `:tabnew` shows an error:
```
E5108: Error executing lua ...te/pack/packer/opt/bufferline.nvim/lua/bufferline/ui.lua:125: Expected lua string
stack traceback:
        [C]: in function 'strwidth'
        ...te/pack/packer/opt/bufferline.nvim/lua/bufferline/ui.lua:125: in function 'get_component_size'
        ...te/pack/packer/opt/bufferline.nvim/lua/bufferline/ui.lua:549: in function <...te/pack/packer/opt/bufferline.nvim/lua/b
ufferline/ui.lua:504>
        vim/shared.lua: in function 'get_components'
        .../site/pack/packer/opt/bufferline.nvim/lua/bufferline.lua:100: in function <.../site/pack/packer/opt/bufferline.nvim/lu
a/bufferline.lua:97>
```

### What did you expect to happen?

Open a new tab.

### Config

```
require("bufferline").setup {
  options = {
    mode = "tabs",
    always_show_bufferline = false,
    show_buffer_icons = false,
    show_buffer_close_icons = false,
    show_close_icon = false,
    modified_icon = "",
    show_tab_indicators = false,
    tab_size = 5,
    numbers = function(opts)
      if vim.api.nvim_get_current_tabpage() == opts.id then return "" end
      return opts.ordinal
    end,
  },
  highlights = {
    indicator_selected = {
      guifg = {
          attribute = "fg",
          highlight = "Type"
      },
      guibg = {
          attribute = "bg",
          highlight = "Type"
      },
    },
    numbers= {
      guifg = {
          attribute = "fg",
          highlight = "Type"
      },
    },
    buffer_selected = {
      gui = "bold",
    },
    pick = {
      gui = "bold",
    },
    pick_selected = {
      gui = "bold",
    },
  },
}
```

### Additional Information

...

### commit

e9697f74612f90cc1fac7b5a973e1aa1323f9aff

**Comment by akinsho** [comment]
@horseinthesky can you please narrow down the commit that causes this issue, I've opened a new tab several times in the last few days and it's worked fine

**Comment by horseinthesky** [comment]
@akinsho I'm on HEAD and I've just printed each segment and one of them has a number in `text` attribute:
```
local function get_component_size(segments)
  assert(vim.tbl_islist(segments), "Segments must be a list")
  local sum = 0
  for _, s in pairs(segments) do
    if has_text(s) then
      print(vim.inspect(s))
      sum = sum + strwidth(s.text)
    end
  end
  return sum
end
```

When doing `:tabnew`:
```
{
  highlight = "BufferLineBackground",
  text = "config.go"
}
{
  text = " "
}
{
  highlight = "BufferLineSeparator",
  text = "▕"
}
{
  highlight = "BufferLineBackground",
  text = " "
}
{
  highlight = "BufferLineBackground",
  text = " "
}
{
  attr = {
    __id = "number"
  },
  highlight = "BufferLineNumbers",
  text = 1
}
```

**Comment by akinsho** [comment]
Looks like all text fields should be explicitly converted to numbers, and it's just erroring because numbers aren't implicitly converted on lua

**Comment by akinsho** [comment]
I don't use numbers so hadn't noticed this error I think

**Comment by horseinthesky** [comment]
Yeah. I guess changing line 125 to ` sum = sum + strwidth(tostring(s.text))` should be fine.
I can do a PR tomorrow if you don't mind.

**Comment by akinsho** [comment]
@horseinthesky that would be great 👍🏾

**Comment by horseinthesky** [comment]
[PR with a fix](https://github.com/akinsho/bufferline.nvim/pull/442)

**Comment by akinsho** [comment]
fixed by #442
